File: GymPlatform/backend/src/main/java/com/gymplatform/config/SystemAccounts.java
package com.gymplatform.config;

import com.gymplatform.domain.entity.Organization;
import com.gymplatform.domain.entity.User;

/** Cuenta bootstrap de pruebas: puede iniciar sesión pero no aparece en listados UI. */
public final class SystemAccounts {

    private SystemAccounts() {}

    public static boolean isBootstrapUser(User user) {
        if (user == null || user.getEmail() == null) {
            return false;
        }
        return isBootstrapEmail(user.getEmail());
    }

    public static boolean isBootstrapEmail(String email) {
        return email != null
                && DefaultAdminCredentials.EMAIL.equalsIgnoreCase(email.trim());
    }

    public static boolean isBootstrapOrganization(Organization org) {
        return org != null
                && DefaultAdminCredentials.ORG_SLUG.equalsIgnoreCase(org.getSlug());
    }

    public static boolean isBootstrapOrganizationId(Long organizationId) {
        return organizationId != null && organizationId.equals(DefaultAdminCredentials.ORG_ID);
    }
}

File: GymPlatform/web/src/utils/systemAccounts.ts
/** Cuenta bootstrap de pruebas: login gymplatformadmin — oculta en listados UI. */
const BOOTSTRAP_EMAIL = 'gymplatformadmin@gymplatform.local'

export function isBootstrapUser(user: { email?: string | null }): boolean {
  return (user.email ?? '').trim().toLowerCase() === BOOTSTRAP_EMAIL
}

export function withoutBootstrapUsers<T extends { email?: string | null }>(users: T[]): T[] {
  return users.filter((u) => !isBootstrapUser(u))
}
